### PR TITLE
APEXMALHAR-2004: Add file's modification time in referenceTimes map instead of parent's

### PR DESCRIPTION
**Pull request by tushargosavi**

### CODE REVIEW

**Comment by tushargosavi** [comment]
@chandnisingh please review.

**Comment by chandnisingh** [comment]
This was done so that the state in memory doesn't become too big. The assumption was that if file is changed the timestamp of the parent directory will change as well.

Are you seeing that times of the file is more recent than the parent?

Storing every file reference in memory may be problematic.

**Comment by otterc** [comment]
@tushargosavi 
Can there be a test case added?
And maybe we need to pull this into previous releases

File: library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java [hunk 1]
```diff
@@ -439,12 +439,12 @@ protected ScannedFileInfo createScannedFileInfo(Path parentPath, FileStatus pare
       ScannedFileInfo info;
       if (rootPath == null) {
         info = parentStatus.isDirectory() ?
-          new ScannedFileInfo(parentPath.toUri().getPath(), childPath.getName(), parentStatus.getModificationTime()) :
-          new ScannedFileInfo(null, childPath.toUri().getPath(), parentStatus.getModificationTime());
+          new ScannedFileInfo(parentPath.toUri().getPath(), childPath.getName(), childStatus.getModificationTime()) :
+          new ScannedFileInfo(null, childPath.toUri().getPath(), childStatus.getModificationTime());
```
Comment (otterc) [inline library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java:443]: please remove `@SuppressWarnings("UnusedParameters")` with `childStatus` in the method definition

Comment (tushargosavi) [inline library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java:443]: done

**Comment by tushargosavi** [comment]
@chandnisingh  I have also added a test case

**Comment by chinmaykolhatkar** [comment]
I see all comments are take care of.
If there are no further comment, I can merge this in a day.

**Comment by otterc** [comment]
@chinmaykolhatkar have you reviewed the pull request or you just want to merge it?

**Comment by chinmaykolhatkar** [comment]
Hi @chandnisingh,

I did review.. maybe I missed something. Apologies for that.
I leave the rest of the review and merging to you.
- Chinmay.

**Comment by otterc** [comment]
@chinmaykolhatkar just wanted to make sure that the changes were reviewed. If you have reviewed then please go ahead with merging

**Pull request merged**